### DISCUSSION TITLE
Webpack JavaScript Code URL Generator

### DISCUSSION THREAD

**Discussion by alessandro-podo** [symfony/symfony#42923] [Q&A] [answered]
Hi there,

so far I have written my JavaScript code in the Twig files and could use the URL generator for the AJAX calls.
Now when I work with Webpack Encore, I can no longer access the URL generator. (or?)

What is the best practice that I always have the URL of the current environment in my JavaScript file?

I'm currently experimenting with Webpack ..

Thanks
Sandro

**Comment by ingfdoaguirre** [comment, marked as answer]
Hi.

There is a section in the Webpack docs in symfony telling about that:

https://symfony.com/doc/current/frontend/encore/server-data.html

Bassically you need to pass data using data attributes, something like this:

`<div id="url" data-url="{{ path("route") }}"></div>`

**Reply by alessandro-podo** [reply]
Thank you for your prompt reply. It was too easy. 
Was expecting a function that makes the variable available as a global variable.

**Reply by chapterjason** [reply]
@alessandro-podo  You can also use this bundle: https://github.com/FriendsOfSymfony/FOSJsRoutingBundle

> This bundle allows you to expose your routing in your JavaScript code. That means you'll be able to generate URL with given parameters like you can do with the Router component provided in the Symfony2 core.
